epel/map.jinja

```diff
@@ -0,0 +1,51 @@
+# Enable by default
+{% set epel={
+  'disabled': False,
+  'key_name': 'RPM-GPG-KEY-EPEL',
+}
+ %}
+
+# specific map for Amazon linux
+{% if grains['os'] == 'Amazon' and grains['osmajorrelease'] in ['2014', '2015', '2016' ]  %}
+  {% set os_map = {
+    'key': 'https://fedoraproject.org/static/0608B895.txt',
+    'key_hash': 'sha256=16d35fa467c6efcee21d3aa068a02054b6a89a7a33bffa94db1fc141693d62a3',
+    'key_name': 'RPM-GPG-KEY-EPEL-6',
+    'rpm': 'http://download.fedoraproject.org/pub/epel/6/i386/epel-release-6-8.noarch.rpm',
+  } %}
+
+{% else %}
+
+# A lookup table for EPEL GPG keys & RPM URLs for various RedHat releases
+  {% set os_map = salt['grains.filter_by']({
+      '5': {
+        'key': 'https://getfedora.org/static/A4D647E9.txt',
+        'key_hash': 'sha256=664c06f579d914f2cf25d05d4d581b8ddec77cae4f72f4020c3a9264b9d5ee71',
+        'key_name': 'RPM-GPG-KEY-EPEL-5',
+        'rpm': 'http://download.fedoraproject.org/pub/epel/5/i386/epel-release-5-4.noarch.rpm',
+      },
+      '6': {
+        'key': 'https://getfedora.org/static/0608B895.txt',
+        'key_hash': 'sha256=16d35fa467c6efcee21d3aa068a02054b6a89a7a33bffa94db1fc141693d62a3',
+        'key_name': 'RPM-GPG-KEY-EPEL-6',
+        'rpm': 'http://download.fedoraproject.org/pub/epel/6/i386/epel-release-6-8.noarch.rpm',
+      },
+      '7': {
+        'key': 'https://getfedora.org/static/352C64E5.txt',
+        'key_hash': 'sha256=22f25ad95d5e8d371760815485dba696ea3002fc2c7f812f2c75276853387107',
+        'key_name': 'RPM-GPG-KEY-EPEL-7',
+        'rpm': 'http://download.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-6.noarch.rpm',
+      },
+    }
+    , grain='osmajorrelease')
+  %}
+{% endif %}
```

Comment:
true, I did it kind of wrong, replacing `'RedHat'` with `'default'` fixes the problem though, since unless it gets a match in Amazon it picks the default mapping, assuming all the other OSes have that `osmajorrelease`.
From the docs [here](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.grains.html): **default** lookup_dict's key used if the grain does not exists or if the grain value has no match on lookup_dict. If unspecified the value is `"default"`.